source/extensions/filters/common/auth/attribute_context.cc

```diff
@@ -0,0 +1,91 @@
+#include "extensions/filters/common/auth/attribute_context.h"
+
+#include "common/http/utility.h"
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Auth {
+
+void AttributeContextUtils::setSourcePeer(envoy::service::auth::v2alpha::AttributeContext& context,
+                                          const Network::Connection& connection,
+                                          const std::string& service) {
+  setPeer(*context.mutable_source(), connection, service, false);
+}
+
+void AttributeContextUtils::setDestinationPeer(
+    envoy::service::auth::v2alpha::AttributeContext& context,
+    const Network::Connection& connection) {
+  setPeer(*context.mutable_destination(), connection, "", true);
+}
+
+void AttributeContextUtils::setPeer(envoy::service::auth::v2alpha::AttributeContext_Peer& peer,
+                                    const Network::Connection& connection,
+                                    const std::string& service, const bool local) {
+
+  auto addr = local ? connection.localAddress() : connection.remoteAddress();
```

Comment:
Please go through and constify local variables. Many of them can be const, won't comment on all of them.